ui-v2/src/components/ui/run-card/run-card.tsx

```diff
@@ -0,0 +1,136 @@
+import { components } from "@/api/prefect";
+import {
+	Breadcrumb,
+	BreadcrumbItem,
+	BreadcrumbList,
+	BreadcrumbSeparator,
+} from "@/components/ui/breadcrumb";
+import { Card } from "@/components/ui/card";
+import { Icon } from "@/components/ui/icons";
+import { StateBadge } from "@/components/ui/state-badge";
+import { TagBadgeGroup } from "@/components/ui/tag-badge-group";
+import { Typography } from "@/components/ui/typography";
+import { secondsToApproximateString } from "@/utilities/seconds";
+import { Link } from "@tanstack/react-router";
+import { cva } from "class-variance-authority";
+import { format, parseISO } from "date-fns";
+
+type Props = {
+	flow: components["schemas"]["Flow"];
+	flowRun: components["schemas"]["FlowRun"];
+	/** If task run is included, uses fields from task run over flow run */
+	taskRun?: components["schemas"]["TaskRun"];
+};
+
+export const RunCard = ({ flow, flowRun, taskRun }: Props) => {
+	const state = taskRun ? taskRun.state : flowRun.state;
+	const startTime = taskRun ? taskRun.start_time : flowRun.start_time;
+	const tags = taskRun ? taskRun.tags : flowRun.tags;
+	const runTime = taskRun
+		? taskRun.estimated_run_time
+		: flowRun.estimated_run_time;
+
+	if (!state) {
+		throw new Error("'state' expected");
+	}
+
+	return (
+		<Card className={stateCardVariants({ state: state.type })}>
+			<div className="flex justify-between items-center">
+				<ConcurrencyLimitTaskRunBreadcrumb
+					flow={flow}
+					flowRun={flowRun}
+					taskRun={taskRun}
+				/>
+				<div>
+					<TagBadgeGroup maxTagsDisplayed={5} tags={tags} />
+				</div>
+			</div>
+			<div className="flex gap-2 items-center text-slate-600">
+				<StateBadge state={state} />
+				{startTime && <StartTime time={startTime} />}
+				<TimeRan duration={runTime} />
+			</div>
+		</Card>
+	);
+};
+
+const ConcurrencyLimitTaskRunBreadcrumb = ({
+	flow,
+	flowRun,
+	taskRun,
+}: Props) => {
+	if (!flow.id || !flowRun.id) {
+		throw new Error("'id' field expected");
+	}
```

Comment:
Task runs aren't guaranteed to have a parent flow run, so we'll need to handle this case without erroring.

Reply:
I think I addressed this in the latest push